lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1462,7 +1463,7 @@ abstract class DefaultTextEditingStrategy with CompositionAwareMixin implements
       if (event.keyCode == _kReturnKeyCode) {
         onAction!(inputConfiguration.inputAction);
         // Prevent the browser from inserting a new line when it's not a multiline input.
-        if (inputConfiguration.inputType is! MultilineInputType) {
+        if (inputConfiguration.inputType is! MultilineInputType && inputConfiguration.inputType is! MultilineNoTextInputType) {
```

Comment:
Small suggestion to simplify this a bit:
```suggestion
        if (inputConfiguration.inputType is! MultilineInputType) {
```
and make it `class MultilineNoTextInputType extends MultilineInputType { ... }`.